### ISSUE TITLE
Add general diagram

### ISSUE THREAD

**Issue by SergioGasquez** [esp-rs/book#76]
We could add a diagram that shows the whole ecosystem in some intro chapters of the book.
@FMFigueroa already created this: https://sharing.clickup.com/9008002852/wb/8cepdt4-184/whiteboard which can be used as a starting point

**Comment by SergioGasquez** [comment]
@f-hollow suggested having 2 diagrams instead of a big one:
- The diagrams would be architecture and approaches
- If we make 2 diagrams it would be easier to include it in the book as the would be smaller

**Comment by SergioGasquez** [comment]
Thanks for working on this! As suggested by @f-hollow I would do 2 different diagrams, the one that you are working on that highlights the difference between architectures and another for `std` vs `no_std`. A few things on your diagram (feel free to address or ignore them):
- I would remove the `esp-idf with rust` block as it will be covered in the other diagram
- Since it will be included in the book, we will already know which modules are supported, so I would also remove the modules sections.
  - Basically, it's a way to make the diagram easier to maintain over time
- In the Xtensa linker I would only put GCC, once we use LLD we would have to change it
  - Atm, it looks like it's a combination of the 2 linkers
- For the targets, I would show both `std` and `no_std` targets in this diagram
